src/window/createStatWidget.ts

```diff
@@ -0,0 +1,72 @@
+import * as events from "../helpers/events";
+import { getElementVisibility } from "../helpers/flexHelper";
+import {
+  WidgetCreator,
+  FlexiblePosition,
+  groupbox,
+  horizontal,
+  label,
+  compute,
+  Store,
+} from "openrct2-flexui";
+
+/**
+ * Creates a stat widget with the given properties.
+ */
+export function createStatWidget<T>(props: {
+  /** The title of your widget which will display in the box*/
+  title: string;
+  /** The store with total game stat values*/
+  gameStatStore: Store<T>;
+  /** The store with this park stat values*/
+  parkStatStore: Store<T>;
+  /**
+   * Function to process the stat value,
+   * e.g. to process time in seconds into nice human-readable text,
+   * or to take the take the length of an array instead of its values*/
+  processStat: (stat: T) => string;
```

Comment:
I feel like renaming this to `formatDisplay` for consistency would make sense. It's also more descriptive.